src/main/java/com/oath/halodb/DBDirectory.java

```diff
@@ -66,9 +77,11 @@ Path getPath() {
      * an exception we silently swallow it.
      */
     void syncMetaData() {
-        try {
-            directoryChannel.force(true);
-        } catch (IOException e) {
+        if (directoryChannel != null) {
+            try {
```

Comment:
Since we are calling `fsync` on the directory only if it is not null (not windows), I think we can remove the try-catch here and let the exception bubble up the call stack.